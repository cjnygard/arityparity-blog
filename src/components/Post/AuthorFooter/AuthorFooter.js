// @flow strict
import React from 'react';
import styles from './AuthorFooter.module.scss';
import Author from './Author';
import Contacts from './Contacts';

type Props = {
  author: {
    name: string,
    bio: string,
    photo: string,
    contacts: {
      [string]: string,
    },
  }
};

const AuthorFooter = ({ author }: Props) => (
  <div className={styles['authorfooter']}>
    <div className={styles['authorfooter__author']}>
      <Author author={author} isIndex={false} />
    </div>
    <div className={styles['authorfooter__contact']}>
      <Contacts contacts={author.contacts} />
    </div>
  </div>
);

export default AuthorFooter;
