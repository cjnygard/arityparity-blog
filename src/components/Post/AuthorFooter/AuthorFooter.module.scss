@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.authorfooter {
  border-top: 1px solid $color-gray-border;
  max-width: $layout-post-width;
  padding-top: 20px;
  @include line-height(1);
  @include margin-top(1);
  @include margin-bottom(2);

  &__author {
    display: block;
    float: left;
    text-decoration: underline;
  }
  &__contact {
    display: block;
    float: right;
    text-decoration: underline;
  }
}

@include breakpoint-sm {
  .authorfooter {
    margin-left: auto;
    margin-right: auto;
  }
}
