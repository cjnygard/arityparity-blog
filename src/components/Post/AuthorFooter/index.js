// @flow strict
export { default } from './AuthorFooter';
