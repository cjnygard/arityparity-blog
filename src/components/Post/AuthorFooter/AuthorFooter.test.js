// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import AuthorFooter from './AuthorFooter';
import siteMetadata from '../../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../../types';

describe('AuthorFooter', () => {
  const props = {
    author: {
      id: 'foobar',
      name: 'test',
      photo: '/foobar-photo-social.jpg',
      bio: 'test',
      contacts: {
        email: 'foobar',
        facebook: 'foobar',
        telegram: 'foobar',
        twitter: 'foobar',
        github: 'foobar',
        rss: 'foobar',
        vkontakte: 'foobar',
        linkedin: 'foobar',
        instagram: 'foobar',
        line: 'foobar',
        gitlab: 'foobar',
        weibo: 'foobar',
        codepen: 'foobar',
        youtube: 'foobar',
        soundcloud: 'foobar'
      }
    },
    isIndex: false
  };

  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(siteMetadata)
      ),
      useStaticQuery.mockReturnValue(siteMetadata)
    );
  });

  it('renders correctly', () => {
    const tree = renderer.create(<AuthorFooter {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
