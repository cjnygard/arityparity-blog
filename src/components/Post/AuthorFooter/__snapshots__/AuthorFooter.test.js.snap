// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorFooter renders correctly 1`] = `
<div
  className="authorfooter"
>
  <p
    className="authorfooter__bio"
  >
    Test bio
    <a
      className="authorfooter__bio-twitter"
      href="https://www.twitter.com/#"
      rel="noopener noreferrer"
      target="_blank"
    >
      <strong>
        Test name
      </strong>
       on Twitter
    </a>
  </p>
</div>
`;
